server.js

```diff
@@ -47,74 +49,188 @@ app.get("/", (req, res) => {
 });
 
 app.post("/login", (req, res) => {
-  Users.findAll()
+  const user_name = req.body.user_name ? req.body.user_name : "";
+  const pass_word = req.body.pass_word ? req.body.pass_word : "";
+  const email = req.body.email ? req.body.email : "";
+
+  Users.findAll({
+    where: Sequelize.or(
+      { user_name, pass_word },
+      Sequelize.and({ pass_word, email })
+    ),
+  })
     .then((journals) => {
-      res.json(journals);
+      // user does not exist or either username, email or password are not valid.
+      if (!journals.length) {
+        res.sendStatus(404);
+      } else {
+        res.json(journals);
+      }
     })
     .catch((err) => {
-      res.send("Error " + err);
+      console.log(err);
+      res.sendStatus(500);
     });
 });
-
+// Not sure if this will be needed
 app.post("/logout", (req, res) => {
   res.send("logout");
 });
 
-app.post("/register", (req, res) => {
+app.post("/register", async (req, res) => {
   const { user_name, email, pass_word } = req.body;
-  //TODO: user_name and email need to be unique
-  /*
-   If the user or email exist send a 409, if another error occurs, send a 500.
-   Send a 200 if everything is ok.
-  */
-  Users.create({ user_name, email, pass_word }).then(()=> {
-    res.sendStatus(200);
-  }).catch(err => {
-    res.send(500);
-  })
+
+  // Checks user_name does not exist
+  const isUsernameInUse = await Users.findAll({ where: { user_name } });
```

Comment:
`isSomething..` is a convention for naming boolean variables, meaning that the variable contains true or false.  In this case `isUsernameInUse` contains an array. Please, rename it to something like `usersWithUserName`.